From 147f140da551f07911789834d07db56ac4233e22 Mon Sep 17 00:00:00 2001
From: djane <d.mabelin@arcanys.com>
Date: Fri, 3 Apr 2020 15:23:12 +0800
Subject: [PATCH 1/3] prepend `path.module` to zip path.

---
 chalice/package.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/chalice/package.py b/chalice/package.py
index 875eb30..f54e325 100644
--- a/chalice/package.py
+++ b/chalice/package.py
@@ -978,8 +978,8 @@ class TerraformCodeLocationPostProcessor(TemplatePostProcessor):
                 asset_path = os.path.join(outdir, 'deployment.zip')
                 self._osutils.copy(r['filename'], asset_path)
                 copied = True
-            r['filename'] = "./deployment.zip"
-            r['source_code_hash'] = '${filebase64sha256("./deployment.zip")}'
+            r['filename'] = "${path.module}/deployment.zip"
+            r['source_code_hash'] = '${filebase64sha256("${path.module}/deployment.zip")}'
 
 
 class TemplateMergePostProcessor(TemplatePostProcessor):
-- 
2.21.0


From bd0721ffac9b1af9eea2502a30b6d000451735a9 Mon Sep 17 00:00:00 2001
From: djane <d.mabelin@arcanys.com>
Date: Fri, 3 Apr 2020 15:34:02 +0800
Subject: [PATCH 2/3] fix flake8 compliance

---
 chalice/package.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/chalice/package.py b/chalice/package.py
index f54e325..9559789 100644
--- a/chalice/package.py
+++ b/chalice/package.py
@@ -979,7 +979,8 @@ class TerraformCodeLocationPostProcessor(TemplatePostProcessor):
                 self._osutils.copy(r['filename'], asset_path)
                 copied = True
             r['filename'] = "${path.module}/deployment.zip"
-            r['source_code_hash'] = '${filebase64sha256("${path.module}/deployment.zip")}'
+            r['source_code_hash'] = \
+                '${filebase64sha256("${path.module}/deployment.zip")}'
 
 
 class TemplateMergePostProcessor(TemplatePostProcessor):
-- 
2.21.0


From 33a67ebc6eb3b7e95cf7943207fd8674352a1f78 Mon Sep 17 00:00:00 2001
From: djane <d.mabelin@arcanys.com>
Date: Fri, 3 Apr 2020 15:56:20 +0800
Subject: [PATCH 3/3] update unit tests

---
 tests/unit/test_package.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/unit/test_package.py b/tests/unit/test_package.py
index 2de63dd..b817cda 100644
--- a/tests/unit/test_package.py
+++ b/tests/unit/test_package.py
@@ -81,9 +81,9 @@ def test_terraform_post_processor_moves_files_once():
         'old-dir.zip', os.path.join('outdir', 'deployment.zip'))
     assert mock_osutils.copy.call_count == 1
     assert template['resource']['aws_lambda_function'][
-        'foo']['filename'] == ('./deployment.zip')
+        'foo']['filename'] == ('${path.module}/deployment.zip')
     assert template['resource']['aws_lambda_function'][
-        'bar']['filename'] == ('./deployment.zip')
+        'bar']['filename'] == ('${path.module}/deployment.zip')
 
 
 def test_template_generator_default():
-- 
2.21.0

